### PR TITLE
change set history clearing agency cache

### PR DESCRIPTION
**Pull request by kvahed**
### Scope & Purpose

*(Please describe the changes in this PR for reviewers)*

- [x] :hankey: Bugfix 
- [ ] :pizza: New feature 
- [ ] :hammer: Refactoring 
- [x] :book: CHANGELOG entry made
- [x] :muscle: The behavior in this PR was *manually tested*
- [x] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [ ] No backports required
- [ ] Backports required for: *(Please specify versions)*

#### Related Information

*(Please reference tickets / specification etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added **Regression Tests**
  - [ ] Added new C++ **Unit Tests**
  - [ ] Added new **integration tests** (i.e. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)

Additionally:

- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

*(Include link to Jenkins run etc)*

### Documentation

> All new Features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the changelog so that
> developers and users have a concise overview. 

- [ ] Added a *Changelog Entry* (referencing the corresponding public or internal issue number)
- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Review by neunhoef — APPROVED** [review]
LGTM

File: arangod/Cluster/AgencyCache.h [hunk 1]
```diff
@@ -127,6 +127,13 @@ class AgencyCache final : public arangodb::Thread {
   change_set_t changedSince(
     std::string const& section, consensus::index_t const& last) const;
   
+  /**
+   * @brief         Clean up planned/current changes before index
+   *
+   * @param index   Done index
+   */
+  void clearChanged(std::string const& section, consensus::index_t const& last);
```
Comment (jsteemann) [inline arangod/Cluster/AgencyCache.h:135]:
```suggestion
   * @brief         Clean up planned/current changes up to including index
   *
   * @param section   "Plan" or "Current"
   * @param index   Done index
   */
  void clearChanged(std::string const& section, consensus::index_t const& index);
```

File: arangod/Cluster/AgencyCache.cpp [hunk 2]
```diff
@@ -570,6 +570,14 @@ void AgencyCache::invokeAllCallbacks() const {
 }
 
 
+void AgencyCache::clearChanged(std::string const& what, consensus::index_t const& done) {
+  std::shared_lock g(_storeLock);
+  decltype(_planChanges)& changes = (what == PLAN) ? _planChanges : _currentChanges;
+  if (changes.size() > 0) {
+    changes.erase(changes.begin(), changes.upper_bound(done));
```
Comment (jsteemann) [inline arangod/Cluster/AgencyCache.cpp:577]:
```suggestion
void AgencyCache::clearChanged(std::string const& what, consensus::index_t const& doneIndex) {
  std::shared_lock g(_storeLock);
  decltype(_planChanges)& changes = (what == PLAN) ? _planChanges : _currentChanges;
  if (!changes.empty()) {
    changes.erase(changes.begin(), changes.upper_bound(doneIndex));
```

**Review by jsteemann — APPROVED** [review]

File: arangod/Cluster/AgencyCache.cpp (hunk 2 above)
Comment (kvahed) [reply arangod/Cluster/AgencyCache.cpp:582]: Oh weija

**Review by neunhoef — APPROVED** [review]
LGTM

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13261/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13262/

**Pull request merged**